File: CodeWars/C#/template copy.cs
/* 
Level: 8 kyu
Link: https://www.codewars.com/kata/57a0e5c372292dd76d000d7e/train/csharp
*/

namespace Solution
{
  public static class Program
  {
    public static string repeatStr(int n, string s)
    {
        string str = "";
        for(int i = 0; i < n; i++){
            str += s;
        }
        return str;
    }
  }
}

/* Top solutions analysis */

namespace Solution
{
  using System;
  using System.Linq;
  public static class Program
  {
    public static string repeatStr(int n, string s)
    {
      return String.Concat(Enumerable.Repeat(s, n));
    }
  }
}

// Use of Enumerable.Repeat() to create the sequence
// Use of the String.Concat() to concatenate the sequence

File: CodeWars/Javascript/template copy.js
/* 
Level: 6 kyuu
Link: https://www.codewars.com/kata/55bf01e5a717a0d57e0000ec/javascript
*/

function persistence(num) {
    let numToArray = (val) => val.toString().split('').map(Number);
    
    let product = numToArray(num);
    let count = 0;
    
    while(product.length > 1) {
      product = product.reduce((acc, val) => acc *= val, 1);
      product = numToArray(product);
      count++;
    }
    
    return count;
}
/* Top solutions analysis */

function persistence(num) {
    var times = 0;
    
    num = num.toString();
    
    while (num.length > 1) {
      times++;
      num = num.split('').map(Number).reduce((a, b) => a * b).toString();
    }
    
    return times;
 }